### PR TITLE
Initial project infrastructure

### PR DESCRIPTION
**Pull request by Miseracle**
Add initial project infrastructure - containers, linter, Travis integration and a simple "Hello-world" app.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    -   id: check-ast
+    -   id: check-yaml
+        exclude: vagga.yaml
+    -   id: check-merge-conflict
+    -   id: check-docstring-first
+    -   id: debug-statements
+    -   id: double-quote-string-fixer
+    -   id: end-of-file-fixer
+    -   id: flake8
+        exclude: /__init__.py
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: I think you shouldn't ignore this

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        args:
+        - --max-line-length=120
+    -   id: name-tests-test
+        args:
+        - --django
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: YAGNI

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    -   id: requirements-txt-fixer
+    -   id: trailing-whitespace
+-   repo: git://github.com/FalconSocial/pre-commit-mirrors-pep257
+    sha: f464d46624a959c6969ce6bcf2cd531cdeefb85a
+    hooks:
+    -   id: pep257
+        exclude: /__init__.py
```
Comment (webknjaz) [inline .pre-commit-config.yaml]: YAGNI

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+language: python
+python: 3.5
```
Comment (webknjaz) [inline .travis.yml]: 3.6

File: README.md [hunk 5]
```diff
@@ -1 +1,7 @@
-# Pyramid-Miseracle
\ No newline at end of file
+# Pyramid-Miseracle
+[![Build Status](https://travis-ci.org/kpi-web-guild/django-girls-blog-Miseracle.svg?branch=master)](https://travis-ci.org/kpi-web-guild/Pyramid-Miseracle)
+
+# My project to learn something about Pyramid etc.
```
Comment (webknjaz) [inline README.md]: `This is ...`

File: main/app.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+
+@view_config(route_name='hello', renderer='string')
+def hello_world(request):
+    """Print 'Hello-world'."""
+    return 'Hello World'
+
+
+if __name__ == '__main__':
+    config = Configurator()
```
Comment (webknjaz) [inline main/app.py]: Wrap this piece of code into function `main()` and call it here.

File: vagga.yaml [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    - !PipConfig
+      dependencies: true
+    - !Py3Requirements requirements.txt
+
+  test:
+    setup:
+    - !Ubuntu trusty
+    - !PipConfig
+      dependencies: true
+    - !Depends .pre-commit-config.yaml
```
Comment (webknjaz) [inline vagga.yaml]: I think, this container doesn't need to depend on this config.

**Review by webknjaz — CHANGES_REQUESTED** [review]
* `main` is too common, normally packages are called after the project
* `__init__.py` isn't required at all under Python 3

File: README.md (hunk 5 above)
Comment (Miseracle) [reply README.md]: Mad English skills :D

**Comment by Miseracle** [comment]
I used "main" as a default name - like "foo", "dummy" etc. Will rename it later, right after specifying this project.

**Comment by Miseracle** [comment]
Added container for pytest tests.

File: main/app.py [hunk 8]
```diff
@@ -11,10 +11,14 @@ def hello_world(request):
...
+    """Runs server."""
+    if __name__ == '__main__':
+        config = Configurator()
+        config.add_route('hello', '/')
+        config.scan()
+        app = config.make_wsgi_app()
+        server = make_server('0.0.0.0', 8080, app)
+        server.serve_forever()
+
+main()
```
Comment (webknjaz) [inline main/app.py]: Here's what I meant:
```python
if __name__ == '__main__':
    main()
```

File: main/app.py [hunk 9]
```diff
@@ -11,10 +11,14 @@ def hello_world(request):
...
-if __name__ == '__main__':
-    config = Configurator()
-    config.add_route('hello', '/')
-    config.scan()
-    app = config.make_wsgi_app()
-    server = make_server('0.0.0.0', 8080, app)
-    server.serve_forever()
+def main():
+    """Runs server."""
+    if __name__ == '__main__':
```
Comment (webknjaz) [inline main/app.py]: this check should be at module level

File: vagga.yaml [hunk 10]
```diff
@@ -41,10 +41,10 @@ commands:
...
     run: pre-commit run --all-files
 
-  #test: !Command
-  #  description: Run tests
-  #  container: test
-  #  run: python3 main/test.py
+  test: !Command
+    description: Run tests
+    container: test
+    run: python3 main/tests.py
```
Comment (webknjaz) [inline vagga.yaml]: python3 -> pytest

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: main/app.py [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+@view_config(route_name='hello', renderer='string')
+def hello_world(request):
+    """Print 'Hello-world'."""
+    return 'Hello World'
+
+
+def main():
+    """Run server."""
+    if __name__ == '__main__':
```
Comment (webknjaz) [inline main/app.py]: this check should be at module level

File: main/app.py [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+        config = Configurator()
+        config.add_route('hello', '/')
+        config.scan()
+        app = config.make_wsgi_app()
+        server = make_server('0.0.0.0', 8080, app)
+        server.serve_forever()
+
+
+main()
```
Comment (webknjaz) [inline main/app.py]: Here's what I meant:
```python
if __name__ == '__main__':
    main()
```

File: vagga.yaml [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+commands:
+  lint: !Command
+    description: Run linters
+    container: lint
+    run: pre-commit run --all-files
+
+  test: !Command
+    description: Run tests
+    container: test
+    run: python3 main/tests.py
```
Comment (webknjaz) [inline vagga.yaml]: python3 -> pytest

**Review by webknjaz — CHANGES_REQUESTED** [review]

File: main/app.py (hunk 12 above)
Comment (Miseracle) [reply main/app.py]: Oh, got it. Looks like an inversed main() method in other programming languages.

**Review by webknjaz — APPROVED** [review]
Feel free to merge this if you checked that it's working.

File: main/app.py (hunk 12 above)
Comment (webknjaz) [reply main/app.py]: It's not inversed. The point is that every time you import some module it's contents are being evaluated. This check ensures that it's the direct run of the module via either `./module.py`, or `python -m module`, or `python ./module.py`. In such case `__name__` var is populated with `'__main__'` value, but it contains the actual name of the module whenever it's being imported from other places via `import` statement.

**Pull request merged**